### PR TITLE
pkp/pkp-lib#2877 fix buildpkg script

### PR DESCRIPTION
**Pull request by bozana**
s. https://github.com/pkp/pkp-lib/issues/2877

### CODE REVIEW

File: tools/buildpkg.sh [hunk 1]
```diff
@@ -106,7 +103,7 @@ node_modules      \
 .postcssrc.js     \
 package.json      \
 webpack.config.js \
-lib/ui-library"
```
Comment (NateWr) [inline tools/buildpkg.sh]: this should still be excluded.

File: tools/buildpkg.sh [hunk 2]
```diff
@@ -122,21 +119,22 @@ git submodule -q update --init --recursive >/dev/null || exit 1
 echo "Done"
 
 echo "Installing composer dependencies:"
+curl -sS https://getcomposer.org/installer | php
 echo -n " - lib/pkp ... "
 cd lib/pkp
-composer.phar update --no-dev
+php ../../composer.phar update --no-dev
```
Comment (NateWr) [inline tools/buildpkg.sh]: I think this should be install actually. It fixed (some) problems for me.

File: tools/buildpkg.sh [hunk 3]
```diff
@@ -122,21 +119,22 @@ git submodule -q update --init --recursive >/dev/null || exit 1
...
 cd ../..
 echo "Done"
 
 echo -n " - plugins/paymethod/paypal ... "
 cd plugins/paymethod/paypal
-composer.phar install --no-dev
+php ../../../composer.phar install --no-dev
```
Comment (NateWr) [inline tools/buildpkg.sh]: Do we normally keep a copy of  `composer.phar` in the root of the project? @asmecher talked about using it from path, but I've always just used it as an executable: `composer install`.

File: tools/buildpkg.sh (hunk 1 above)
Comment (bozana) [reply tools/buildpkg.sh]: yes, ok :-)

File: tools/buildpkg.sh (hunk 2 above)
Comment (bozana) [reply tools/buildpkg.sh]: ok

File: tools/buildpkg.sh (hunk 3 above)
Comment (bozana) [reply tools/buildpkg.sh]: I believe that now, that we have several directories/paths it is ok to have composer.phar in the root of the project -- it is the same in the README :-)
But, when you just use composer install, do you need composer.php?

**Comment by bozana** [comment]
@NateWr, I excluded the lib/ui-library again and changed update with install... THANKS!!! Will also test further...

**Comment by NateWr** [comment]
@kaschioudi is looking at the composer issue. Maybe he can chime in here and we can put it all in this PR.

File: tools/buildpkg.sh (hunk 3 above)
Comment (NateWr) [reply tools/buildpkg.sh]: @asmecher we still need your input on how to reference composer. See the comments above.

**Comment by bozana** [comment]
Shall I then include the lib/pkp/lib/vendor/composer folder? I'll try...

**Comment by NateWr** [comment]
Yeah I think so.

**Comment by bozana** [comment]
Hmm... Do you know what is needed from that folder? E.g. this file autoload_static.php requires also some other i.e. ../kriswallsmith/assetic/src/functions.php that are excluded as well...

**Comment by NateWr** [comment]
Yeah I don't know. I think @asmecher or @kaschioudi will need to chime in.

**Comment by kaschioudi** [comment]
@bozana @NateWr 
When I removed `lib/pkp/lib/vendor/composer` from excluded files/folder list I also got an error about `lib/pkp/lib/vendor/kriswallsmith/...` then `lib/pkp/lib/vendor/symfony/...`. So I am confused as well. 

Anybody know the reasons why these folders are excluded manually?  Would it help to move these dependencies from `require` section to `require-dev` or may be use `composer remove` to remove some components before creating the tar.gz archive? @asmecher

**Comment by asmecher** [comment]
Those directories would've been excluded because they were required by Composer (as dependencies) but never used in practice. It's likely that some of these will now be needed in production (e.g. because of the query builder), so feel free to re-add them as needed.

**Comment by asmecher** [comment]
It's possible that moving to an autoloader https://github.com/pkp/pkp-lib/issues/2559 (rather than importing dependencies manually) has made it so that we can't arbitrarily exclude directories any more.

**Comment by asmecher** [comment]
This got resolved elsewhere.

**Pull request closed without merging**